lib/utils.js

```diff
@@ -537,6 +537,32 @@ exports.getError = function (err) {
   return err || exports.undefinedError();
 };
 
+/**
+ * Show a deprecation warning. Each distinct message is only displayed once.
+ *
+ * @param {string} msg
+ */
+exports.deprecate = function (msg) {
+  msg = String(msg);
+  if (seenDeprecationMsg.hasOwnProperty(msg)) {
+    return;
+  }
+  seenDeprecationMsg[msg] = true;
+  doDeprecationWarning(msg);
+};
+
+var seenDeprecationMsg = {};
+
+var doDeprecationWarning = process.emitWarning
+  // Node.js v6+
+  ? function (msg) { process.emitWarning(msg, 'DeprecationWarning'); }
+  // Node.js v4+
+  // process.emitWarning emits on the next tick; emulate that.
+  : process.nextTick
+    ? function (msg) { process.nextTick(() => console.warn(msg)); }
+    // Everything else
+    : function (msg) { Promise.resolve().then(() => console.warn(msg)); };
```

Comment:
This isn't necessary; we can assume `process.nextTick` will always be defined.  The browser gets a polyfill.  We can safely remove this last branch of the ternary.

Reply:
Done, and rebased again.